### ISSUE TITLE
librivox unsupervised dataset

### ISSUE THREAD

**Issue by lunixbochs** [facebookresearch/wav2letter#491]
Hi, can you provide more information on the "with generated pseudo labels" part of the librivox unsupervised dataset generation? I assume you ran the Decode step on your VAD sections from libri-lite with one of the new SOTA models. Have you released the acoustic and language models used to do this?

I'm planning to test my training harness on the word-piece models, but I don't think I have enough information to reproduce the unsupervised librivox part yet.

Can you also advise on the "trained for 2-4 days" part? How many V100 GPUs were involved? 60k hours is quite a bit to retrain independently so it will help me know what to expect :)

**Comment by avidov** [comment]
Hi,

I think that all of your questions are answered in our [SELF-TRAINING FOR END-TO-END SPEECH RECOGNITION](https://arxiv.org/pdf/1909.09116.pdf) paper.

**Comment by jacobkahn** [comment]
@lunixbochs — the paper @avidov mentioned above describes the base recipe for what we're doing. I haven't yet released a recipe for the self-training work, but I hope to do it in the next week now that I'm back. Your description is correct - we used slightly different segmentations of the LibriLight data to generate the pseudo-labels, but that shouldn't make much of a difference.

We've just released models for the [recent pseudo-label work](https://arxiv.org/abs/1911.08460) — the recipe (and links to pretrained models) [is here](https://github.com/facebookresearch/wav2letter/tree/master/recipes/models/sota/2019). We didn't release the pseudo-labels themselves, but as described in the paper, we use a transformer + CTC model to generate them. The [LibriVox section](https://github.com/facebookresearch/wav2letter/tree/master/recipes/models/sota/2019/librivox) of the recipe also describes how long these models were trained for. In general, LibriVox models handily beat the LibriSpeech models relatively quickly, but took a while to achieve sub 5.0 dev performance without decoding. The epochs described in that readme are epochs over the entire LibriVox set; unless otherwise specified, models were trained on 64 V100s. So yes, the scale's quite large and things trained for a long time.
